src/fileio/datafile.cxx

```diff
@@ -539,6 +541,8 @@ bool Datafile::write(const char *format, ...) const {
   tmp.openw(filename);
   bool ret = tmp.write();
   tmp.close();
+
+  delete[] filename;
```

Comment:
If you `delete[] filename` here, does the destructor need to do it as well? I think here is probably the better place.